retrieval/discovery/marathon/conversion.go

```diff
@@ -24,18 +22,25 @@ func AppsToTargetGroups(apps *AppList) map[string]*config.TargetGroup {
 func createTargetGroup(app *App) *config.TargetGroup {
 	var (
 		targets = targetsForApp(app)
-		source  = targetGroupName(app)
+		source  = sanitizeName(app.ID)
```

Comment:
Source can be any string really. But the sanitization below could theoretically sanitize distinct IDs into the same source. So let's drop it.